beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -571,6 +637,58 @@ func (f *blocksFetcher) fetchBlobsFromPeer(ctx context.Context, bwb []blocks2.Bl
 	return nil, errNoPeersAvailable
 }
 
+// fetchColumnsFromPeer fetches blocks from a single randomly selected peer.
+func (f *blocksFetcher) fetchColumnsFromPeer(ctx context.Context, bwb []blocks2.BlockWithROBlobs, pid peer.ID, peers []peer.ID) ([]blocks2.BlockWithROBlobs, error) {
+	ctx, span := trace.StartSpan(ctx, "initialsync.fetchColumnsFromPeer")
+	defer span.End()
+	if slots.ToEpoch(f.clock.CurrentSlot()) < params.BeaconConfig().DenebForkEpoch {
+		return bwb, nil
+	}
+	columnWindowStart, err := prysmsync.DataColumnsRPCMinValidSlot(f.clock.CurrentSlot())
+	if err != nil {
+		return nil, err
+	}
+	// Construct request message based on observed interval of blocks in need of columns.
+	req := countCommitments(bwb, columnWindowStart).blobRange(f.bs).RequestDataColumns()
+	if req == nil {
+		return bwb, nil
+	}
+	// Construct request message based on required custodied columns.
+	custodyColumns, err := peerdas.CustodyColumns(f.p2p.NodeID(), params.BeaconConfig().CustodyRequirement)
+	if err != nil {
+		return nil, err
+	}
+	var colIdxs []uint64
```

Comment:
Could be a `make` with fixed size/capacity to avoid possible possible eventual re-allocation.